### PR TITLE
Tried to improve #165. Chapter 8.2 Challenges /C120/

### LINKED ISSUE #165: Discussion Improvements
/C120/   Point out mocking, as it is not clear if we meant internal or external calls.
Components -> PCM instances

### PR DESCRIPTION
**Pull request by flyfl**
Fix #165. I did not know how to change:
>Components -> PCM instances

That's because I don't see any reference to PCM. We don't want to measure an instance of a PCM.

### CODE REVIEW

File: doc/Requirements Specification/Parts/Discussion.lyx [hunk 1]
```diff
@@ -385,22 +359,9 @@ gls{component}
...
-\backslash
-glsuserii{component}
-\end_layout
-
-\end_inset
-
- behaviour can be done automatically needs investigation.
- Such imitations may not affect the measurement results.
+ might be very effortful or even impossible.
+ If this problem can't be solved, Beagle should be restricted on analysing
```
Comment (acBerger) [inline doc/Requirements Specification/Parts/Discussion.lyx]: "cannot" instead of "can't"

**Comment by acBerger** [comment]
[PDF] (http://flyfl.github.io/Beagle/Requirements%20Specification.pdf)

**Comment by m1cm1c** [comment]
Server software doesn't necessarily mean that the software terminates after its done. A better restriction would be software which terminates without user interaction.

**Comment by flyfl** [comment]
Do you mean software componets instead of "software". If so, I would agree.

**Comment by m1cm1c** [comment]
You currently say "If this problem cannot be solved, Beagle should be restricted on analysing server software." so I stuck to this. But "software components" would be better.

**Comment by jGleitz** [comment]
[Krogmann 2011] solves this as follows:

> Furthermore, Beagle requires the availability of a test bed in which the software system under study can be executed. This can either be a fully running software system or a software system where mock-ups realise required functionality.

**Comment by RomanLangrehr** [comment]
> But "software components" would be better.

I agree.

But why do we restict at all? Can't we just say "The user has to ensure, that all components are mocket, that need to be mocked."?!

**Comment by jGleitz** [comment]
> But why do we restict at all? Can't we just say "The user has to ensure, that all components are mocket, that need to be mocked."?!

That’s what Krogmann did (see above). Although his wording was more elegant ;)

**Comment by flyfl** [comment]
This point may belong into "Boundary", if we state out that Beagle will not mock any components.

**Comment by RomanLangrehr** [comment]
> This point may belong into "Boundary", if we state out that Beagle will not mock any components.

I disagree. This point is too detailed for the purpose chapter. I think we should change it here.

**Comment by jGleitz** [comment]
I agree with @RomanLangrehr

**Comment by flyfl** [comment]
Please review

File: doc/Requirements Specification/Parts/Discussion.lyx [hunk 2]
```diff
@@ -357,21 +344,7 @@ glspl{component}
...
-
-
-\backslash
-gls{component}
-\end_layout
-
-\end_inset
-
- calls another 
+ (or bundles) at a time, some internal
```
Comment (jGleitz) [inline doc/Requirements Specification/Parts/Discussion.lyx]: it’s unclear what “bundles” refers to here. How about changing the sentence to “As Beagle should be able to measure specific components” (you can omit “at a time”, I think)

Comment (jGleitz) [inline doc/Requirements Specification/Parts/Discussion.lyx]: There is no such thing as an “internal component”. How about “[…] other components that are called during the measured component’s execution may wished to be mocked”.

File: doc/Requirements Specification/Parts/Discussion.lyx [hunk 3]
```diff
@@ -399,22 +373,9 @@ gls{component}
...
-glsuserii{component}
-\end_layout
-
-\end_inset
-
- behaviour can be done automatically needs investigation.
- Such imitations may not affect the measurement results.
+ might be very effortful or even impossible.
+ If this problem cannot be solved, the user has to ensure, that all of these
+ are mocket, which need to be.
```
Comment (jGleitz) [inline doc/Requirements Specification/Parts/Discussion.lyx]: I don’t quite like that wording. How about:

“Beagle may require the user to provide a test bed in which the measured components can be executed.”

File: doc/Requirements Specification/Parts/Discussion.lyx (hunk 2 above)
Comment (flyfl) [inline doc/Requirements Specification/Parts/Discussion.lyx]: I wanted to distinguish between internal calls (that happen within the software/software components) and external calls (e.g. a call that will open another software), that are not interesting for us

Comment (jGleitz) [inline doc/Requirements Specification/Parts/Discussion.lyx]: Well any call to another component has do work, doesn’t it? So why distinguish?

Comment (flyfl) [inline doc/Requirements Specification/Parts/Discussion.lyx]: Of course the depended components of the specific software have to work. But not the execution of different software (e.g. PDF Reader wants to open URL in a Browser). Maybe this distiction is negligible for us, but I remeber Michael, telling us something like that (Friday meeting)

Comment (jGleitz) [inline doc/Requirements Specification/Parts/Discussion.lyx]: well, whatever. “internal” definitely makes no sense. So what do you think of “[…] other components in the software that are called during the measured component’s execution may wished to be mocked”

**Comment by flyfl** [comment]
I can't do anything in my SmartGit repository as an unfriedly error occured. I would suggest to merge this PR and if I get my smartGit to work again, I will adopt your changes.

**Comment by jGleitz** [comment]
We still have some hours. Let’s not panic and merge this when it is ready.

**Comment by jGleitz** [comment]
What will happen to the issues mentioned in the above comments?

**Comment by flyfl** [comment]
I will create a new PR. I have trouble to access my origin repository. When I'm done, I will open a new PR.

**Pull request merged**